docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,276 @@
+# How to integrate upstream methods in your library
+
+Now that you've seen more downstream utilities for downloading files, it is time to introduce you to additional *upstream* utilities. These are utilities that publish files to the Hub from your library. This guide will show you how to:
+
+* Use the `HfApi` class to manage a repository.
+* Use the `Repository` class to handle files and version control a repository with Git-like commands.
+
+## `HfApi`
+
+The `HfApi` class is a high-level class that wraps around HTTP requests. There are many valuable tasks you can accomplish with the `HfApi` class, including: 
+
+- List and filter models and datasets.
+- Inspect model or dataset metadata.
+- Delete a repository.
+- Change the visibility of a repository.
+
+### List and filter
+
+It can be helpful for users to see a list of available models and filter them according to a specific language or framework. This can be especially useful for library and organization owners who want to view all their models. Use the `list_models` function with the `filter` parameter to search for specific models.
```

Comment:
```suggestion
It can be helpful for users to see a list of available models and filter them according to a specific language or library. This can be especially useful for library and organization owners who want to view all their models. Use the `list_models` function with the `filter` parameter to search for specific models.
```